core/src/main/java/org/apache/carbondata/core/datamap/Segment.java

```diff
@@ -257,4 +270,56 @@ public void setFilteredIndexShardName(String filteredIndexShardName) {
   public LoadMetadataDetails getLoadMetadataDetails() {
     return loadMetadataDetails;
   }
+
+  public String getIndexSize() {
+    return indexSize;
+  }
+
+  public void setIndexSize(String size) {
+    indexSize = size;
+  }
+
+  public ReadCommittedScope getReadCommittedScope() {
+    return readCommittedScope;
+  }
+
+  public boolean isCacheable() {
+    return isCacheable;
+  }
+
+  public void setCacheable(boolean cacheable) {
+    isCacheable = cacheable;
+  }
+
+  @Override public void write(DataOutput out) throws IOException {
+    out.writeUTF(segmentNo);
```

Comment:
Some of the objects are heavy objects (like LoadMetadataDetails) serializing and deserializing those object will take time and it will impact the performance